### PR TITLE
Add health endpoint to Node-RED server

### PR DESCRIPTION
**Pull request by youming-lin**
Add health check endpoint to server, always

### CODE REVIEW

**Comment by knolleary** [comment]
This is the right general approach for solving the question of adding the `/health` endpoint, but you don't need to copy in the whole of the code quite like this. A lot of it is handling command-line arguments - which is completely irrelevant for the starter kit. It can be trimmed back considerably.

**Comment by youming-lin** [comment]
@knolleary I agree. My take on a proper solution would be to essentially take that code and rewrite it into just the bits and pieces we need (may also refactor index.js as well), though admittedly I'm not familiar with the code.

We also talked about having the node-red lib set up the health check endpoint itself, though I was personally against it. Do you think this is the better solution?

**Comment by knolleary** [comment]
Where is this discussion happening? As the originator of the Node-RED starter kit, I'd like to be in the loop on discussions that affect it.

The code exists the way it does because it was the most efficient way to bootstrap a node-red app with a custom getting-started set of screens. I'm not adversed to it being refactored, but there needs to be a clear purpose for doing that refactoring - a clear problem to solve.

I suggest starting by stripping out all the command-line parsing logic in `red.js` within this PR. I can then help refine it further.

**Comment by youming-lin** [comment]
Just within the team right now. I hit this issue, did this quick implementation, brought it up to team, they offered a different idea (I @'ed you in the internal GHE issue on our board).

I'm for improving this PR, as I don't think adding health check to node-red lib adds much value to other users of the lib, and it creates an implicit dependency between the lib and the DevOps toolchain template (toolchain deployment failing health check stage is why I'm adding the health endpoint; there are other solutions besides this but felt this was the right solution). It's also faster to make the change here since the node-red lib probably has its own lengthy process to release a new version.

As for the refactoring, it didn't make much sense to me to keep the red.js file as a script as we are not consuming it via CLI, might as well turn it into an exportable module and/or include in index.js itself.

**Comment by youming-lin** [comment]
@knolleary I've made more changes. Can you review again?

**Review by knolleary — CHANGES_REQUESTED** [review]
Looks good - just one more bit to remove then its :+1: from me.

File: red.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+  require('./routers/health')(app);
+
+  if (process.env.NODE_RED_ENABLE_SAFE_MODE && !/^false$/i.test(process.env.NODE_RED_ENABLE_SAFE_MODE)) {
+    settings.safeMode = true;
+  }
+  if (process.env.NODE_RED_ENABLE_PROJECTS) {
+    settings.editorTheme = settings.editorTheme || {};
+    settings.editorTheme.projects = settings.editorTheme.projects || {};
+    settings.editorTheme.projects.enabled = !/^false$/i.test(process.env.NODE_RED_ENABLE_PROJECTS);
+  }
```
[the remark this answers is not in the document]
Comment (youming-lin) [reply red.js]: I've removed the code block.

**Comment by youming-lin** [comment]
@knolleary I have removed that `if` block now. Can you review again?

**Review by knolleary — APPROVED** [review]

**Pull request merged**